src/core/thread/mle.cpp

```diff
@@ -2918,12 +2918,6 @@ otError Mle::HandleLeaderData(const Message &aMessage, const Ip6::MessageInfo &a
     {
         if (IsChild())
         {
-#if OPENTHREAD_FTD
-            // An FTD skips handling LeaderData of a different partition.
-            VerifyOrExit(!IsFullThreadDevice() || (leaderData.GetPartitionId() == mLeaderData.GetPartitionId() &&
-                                                   leaderData.GetLeaderRouterId() == GetLeaderId()),
-                         error = OT_ERROR_DROP);
-#endif
```

Comment:
We know that `HandeLeaderData()` is called from 3 places:
- `Mle::HandleDataResponse()`
- `Mle::HandleChildUpdateRequest()`
- `Mle::HandleChildUpdateResponse`

This PR is aimed to fix the initial child sync (i.e. Child Update msgs). Just want us to make sure and verify that in other uses of the method, it is still safe/fine to  remove the partition and id check?

Reply:
It's very thoughtful of you. 

Note that this code removed by this PR is added by https://github.com/openthread/openthread/pull/4860 to fix assertion failures when `mRouterTable.GetLeader() == nullptr`. So, this PR restores OT back to the original behavior that OT had  always been so until https://github.com/openthread/openthread/pull/4860, which I believe is safe. 

Thoughts?

Reply:
Thanks for the info/background on this. Sounds all good to me. 👍